src/backend/palrange.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+#include "../ds/address.h"
+#include "../pal/pal.h"
+
+namespace snmalloc
+{
+  template<SNMALLOC_CONCEPT(ConceptPAL) PAL>
+  class PalRange
+  {
+  public:
+    class State
+    {
+    public:
+      PalRange* operator->()
+      {
+        static PalRange range{};
```

Comment:
I guess this is some C++ism? Maybe worth a comment to at least name the pattern being used?